src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -384,6 +384,32 @@ case object EXTCODECOPY extends OpCode(0x3c, 4, 0, G_extcode) {
   }
 }
 
+case object BLOCKHASH extends OpCode(0x40, 1, 1, G_blockhash) with ConstGas {
+  protected def exec[W <: WorldStateProxy[W, S], S <: Storage[S]](state: ProgramState[W, S]): ProgramState[W, S] = {
+    val (blockNumber, stack1) = state.stack.pop
+
+    val hash = if (state.context.env.blockHeader.number - blockNumber > 256 || blockNumber > state.context.env.blockHeader.number) {
```

Comment:
While looking at how other clients implement this opcode, I think I found a possible border case, if `blockNumber == state.context.env.blockHeader.number` we currently fetch the block hash but in EthereumJ it returns `DataWord(0)` in this case (https://github.com/ethereum/ethereumj/blob/master/ethereumj-core/src/main/java/org/ethereum/vm/program/Program.java#L654). Maybe the condition should be `blockNumber >= state.context.env.blockHeader.number`?